web/tests/compose_pm_pill.test.js

```diff
@@ -153,7 +153,9 @@ run_test("pills", ({override}) => {
         callback();
     };
 
-    compose_pm_pill.initialize();
+    compose_pm_pill.initialize({
+        on_pill_create_or_remove: compose_recipient.update_placeholder_text,
+    });
```

Comment:
@andersk just a note on coding patterns -- it seems bad to have this bit of code copy/pasted into individual tests -- it seems like the kind of thing that might just end up diverging. I don't have an easy answer for how to avoid it... but I suppose it's possible that the best right answer would be to just past a `noop` function in these tests rather than trying to pass the hook that we use in production; I'd be kinda surprised if most of these tests did anything but mock the action of `compose_recipient.update_placeholder_text` anyway

Reply:
Yeah, the ideal structure would be that we take advantage of these callbacks to let us test components using natural interfaces without haphazardly mocking their recursive dependencies. The test can pass noop callbacks, or callbacks that assert they’re called with the right arguments, etc.

It’s not going to be easy to transform our cyclic spaghetti blob into a structure that fully supports this, because the whole problem with a cyclic spaghetti blob is that it resists being transformed into anything, but it’s good to identify steps we can take in that direction.